ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3202,7 +3202,7 @@ public void loadTable(Path loadPath, String tableName, LoadFileType loadFileType
         //for fullAcid we don't want to delete any files even for OVERWRITE see HIVE-14988/HIVE-17361
         boolean isSkipTrash = MetaStoreUtils.isSkipTrash(tbl.getParameters());
         boolean needRecycle = !tbl.isTemporary()
-                && ReplChangeManager.shouldEnableCm(Hive.get().getDatabase(tbl.getDbName()), tbl.getTTable());
```

Comment:
This is done at 2 more places in the same class. We should fix that as well?

Reply:
Fixed in recent commit. Fine with the invocation for PartitionIterable as "new PartitionIterable(Hive.get()" (since it could have pointed earlier to diff db).

Reply:
Had to revert the changes in other place.